cmd/collector/app/server/http_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/uber/jaeger-lib/metrics/metricstest"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/collector/app/handler"
+	"github.com/jaegertracing/jaeger/pkg/healthcheck"
+)
+
+// test wrong port number
+func TestFailToListenHttp(t *testing.T) {
+	logger, _ := zap.NewDevelopment()
+	server, err := StartHTTPServer(&HTTPServerParams{
+		HostPort: ":-1",
+		Logger:   logger,
+	})
+	assert.Nil(t, server)
+	assert.EqualError(t, err, "listen tcp: address -1: invalid port")
+}
+
+func TestSpanCollectorHttp(t *testing.T) {
+	logger, _ := zap.NewDevelopment()
+	params := &HTTPServerParams{
+		Handler:        handler.NewJaegerSpanHandler(logger, &mockSpanProcessor{}),
+		SamplingStore:  &mockSamplingStore{},
+		MetricsFactory: metricstest.NewFactory(time.Hour),
+		HealthCheck:    healthcheck.New(),
+		Logger:         logger,
+	}
+
+	server := &http.Server{Addr: ":0"}
```

Comment:
Shouldn't this be created _after_ the listener with `Addr: listener.Addr().String()`? Otherwise it's confusing, looks like two different ports are being used.